Source/JavaScriptCore/runtime/ISO8601.cpp

```diff
@@ -530,10 +530,70 @@ std::optional<int64_t> parseUTCOffsetInMinutes(StringView string)
 template<typename CharacterType>
 static bool canBeCalendar(const StringParsingBuffer<CharacterType>& buffer)
 {
-    // https://tc39.es/proposal-temporal/#prod-Calendar
-    // Calendar :
-    //     [u-ca= CalendarName]
-    return buffer.lengthRemaining() >= 6 && buffer[0] == '[' && buffer[1] == 'u' && buffer[2] == '-' && buffer[3] == 'c' && buffer[4] == 'a' && buffer[5] == '=';
+    // https://tc39.es/proposal-temporal/#sec-temporal-parseisodatetime
+    // Step 4(a)(ii)(2)(a):
+    //  Let key be the source text matched by the AnnotationKey Parse Node contained within annotation
+
+    // https://tc39.es/proposal-temporal/#prod-Annotation
+    // Annotation :::
+    //     [ AnnotationCriticalFlag_opt AnnotationKey = AnnotationValue ]
+
+    // https://tc39.es/proposal-temporal/#prod-AnnotationCriticalFlag
+    // AnnotationCriticalFlag :::
+    //     !
+
+    // https://tc39.es/proposal-temporal/#prod-AnnotationKey
+    // AnnotationKey :::
+    //     AKeyLeadingChar
+    //     AnnotationKey AKeyChar
+
+    // https://tc39.es/proposal-temporal/#prod-AKeyLeadingChar
+    // AKeyLeadingChar :::
+    //     LowercaseAlpha
+    //     _
+
+    // https://tc39.es/proposal-temporal/#prod-AKeyChar
+    // AKeyChar :::
+    //     AKeyLeadingChar
+    //     DecimalDigit
+    //     -
+
+    // This just checks for '[', followed by an optional '!' (critical flag),
+    // followed by a valid key, followed by an '='.
+
+    size_t length = buffer.lengthRemaining();
+    if (!length)
+        return false;
```

Comment:
If BNF is right, then it must have `[`, `=`, `]`, `AnnotationKey` and `AnnotationValue`, right?
Then, what we should check is,

```
// Because of `[`, `=`, `]`, `AnnotationKey` and `AnnotationValue`, annotation must be >= 5.
if (length < 5)
    return false;

 if (*buffer != '[')
    return false;

size_t index = 1;
if (buffer[index] == '!')
    ++index;

...
```